lib/Base/Tensor.cpp

```diff
@@ -543,20 +543,68 @@ void Tensor::init(InitKind init, float val, PseudoRNG &PRNG) {
 }
 
 void Tensor::convertToType(ElemKind newTy) {
-  Tensor tmp(newTy, dims());
-  switch (newTy) {
-  case ElemKind::Float16Ty:
-    assert(getElementType() == ElemKind::FloatTy && "Cast not implemented");
-    tmp.copyWithCast<float16_t, float>(this);
-    break;
-  case ElemKind::FloatTy:
-    assert(getElementType() == ElemKind::Float16Ty && "Cast not implemented");
-    tmp.copyWithCast<float, float16_t>(this);
-    break;
-  default:
-    llvm_unreachable("Type not supported");
+  *this = this->getCopyConvertedToType(newTy);
+}
+
+Tensor Tensor::getCopyConvertedToType(ElemKind newKind) const {
+  if (!isQuantizedElemKind(newKind)) {
+    Tensor tmp(newKind, dims());
+    switch (newKind) {
+    case ElemKind::Float16Ty:
+      assert(getElementType() == ElemKind::FloatTy && "Cast not implemented");
+      tmp.copyWithCast<float16_t, float>(this);
+      break;
+
+    case ElemKind::FloatTy:
+      assert(getElementType() == ElemKind::Float16Ty && "Cast not implemented");
+      tmp.copyWithCast<float, float16_t>(this);
+      break;
+
+    default:
+      llvm_unreachable("Type not supported");
+    }
+    return tmp;
+  }
+
+  // Handle Fused conversion. Currently only supports UInt8FusedQTy ->
+  // UInt8FusedFP16QTy.
+  assert((getElementType() == ElemKind::UInt8FusedQTy) &&
+         (newKind == ElemKind::UInt8FusedFP16QTy) &&
+         "Conversion not implemented.");
+  Tensor tmp(newKind,
+             {dims()[0], dims()[1] - 2 * (sizeof(float) - sizeof(float16_t))},
```

Comment:
can we make an assumption here that dims().size() == 2?

Reply:
There are asserts in other places checking this for the Fused ElemKinds, but I can add a new check for it here as well.